## Question 7 -
Using the data from Question 4, write code to analyze the data and answer the following questions Note -
1. Draw plots to demonstrate the analysis for the following questions for better visualizations
2. Write code comments wherever required for code understanding

Insights to be drawn -
- Get all the Earth meteorites that fell before the year 2000
- Get all the earth meteorites co-ordinates who fell before the year 1970
- Assuming that the mass of the earth meteorites was in kg, get all those whose mass was more
than 10000kg

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:
df = pd.read_csv("Dataset/nasa_meteroite_converted_json_csv.csv")
df.head()

id  :@computed_region_cbhk_fwbd      name  :@computed_region_nnqa_25f4  \
0    1                          NaN    Aachen                          NaN   
1    2                          NaN    Aarhus                          NaN   
2    6                          NaN      Abee                          NaN   
3   10                          NaN  Acapulco                          NaN   
4  370                          NaN   Achiras                          NaN   

   fall  geolocation_coordinates_1  geolocation_coordinates_2  \
0  Fell                    6.08333                   50.77500   
1  Fell                   10.23333                   56.18333   
2  Fell                 -113.00000                   54.21667   
3  Fell                  -99.90000                   16.88333   
4  Fell                  -64.95000                  -33.16667   

  geolocation_type      mass nametype     recclass    reclat    reclong  \
0            Point      21.0    Valid           L5  50.77500    6.08333   
1            Point     720.0    Valid           H6  56.18333   10.23333   
2            Point  107000.0    Valid          EH4  54.21667 -113.00000   
3            Point    1914.0    Valid  Acapulcoite  16.88333  -99.90000   
4            Point     780.0    Valid           L6 -33.16667  -64.95000   

                      year  
0  1880-01-01T00:00:00.000  
1  1951-01-01T00:00:00.000  
2  1952-01-01T00:00:00.000  
3  1976-01-01T00:00:00.000  
4  1902-01-01T00:00:00.000

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 14 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           1000 non-null   int64  
 1   :@computed_region_cbhk_fwbd  133 non-null    float64
 2   name                         1000 non-null   object 
 3   :@computed_region_nnqa_25f4  134 non-null    float64
 4   fall                         1000 non-null   object 
 5   geolocation_coordinates_1    988 non-null    float64
 6   geolocation_coordinates_2    988 non-null    float64
 7   geolocation_type             988 non-null    object 
 8   mass                         972 non-null    float64
 9   nametype                     1000 non-null   object 
 10  recclass                     1000 non-null   object 
 11  reclat                       988 non-null    float64
 12  reclong                      988 non-null    float64
 13  year               

In [37]:
df['year']=df['year'].str.split('-').str[0]
df['year'] = df['year'].fillna("0")
df['year'] = df['year'].astype('int64')

## 1, Get all the Earth meteorites that fell before the year 2000

In [39]:
df[df['year']<2000]

id  :@computed_region_cbhk_fwbd       name  \
0        1                          NaN     Aachen   
1        2                          NaN     Aarhus   
2        6                          NaN       Abee   
3       10                          NaN   Acapulco   
4      370                          NaN    Achiras   
..     ...                          ...        ...   
994  24004                          NaN   Timochin   
995  24009                          NaN   Tirupati   
997  24011                          NaN      Tjabe   
998  24012                          NaN   Tjerebon   
999  24019                          NaN  Tomakovka   

     :@computed_region_nnqa_25f4  fall  geolocation_coordinates_1  \
0                            NaN  Fell                    6.08333   
1                            NaN  Fell                   10.23333   
2                            NaN  Fell                 -113.00000   
3                            NaN  Fell                  -99.90000   
4                            NaN  Fell                  -64.95000   
..                           ...   ...                        ...   
994                          NaN  Fell                   35.20000   
995                          NaN  Fell                   79.41667   
997                          NaN  Fell                  111.53333   
998                          NaN  Fell                  106.58333   
999                          NaN  Fell                   34.76667   

     geolocation_coordinates_2 geolocation_type      mass nametype  \
0                     50.77500            Point      21.0    Valid   
1                     56.18333            Point     720.0    Valid   
2                     54.21667            Point  107000.0    Valid   
3                     16.88333            Point    1914.0    Valid   
4                    -33.16667            Point     780.0    Valid   
..                         ...              ...       ...      ...   
994                   54.50000            Point   65500.0    Valid   
995                   13.63333            Point     230.0    Valid   
997                   -7.08333            Point   20000.0    Valid   
998                   -6.66667            Point   16500.0    Valid   
999                   47.85000            Point     600.0    Valid   

        recclass    reclat    reclong  year  
0             L5  50.77500    6.08333  1880  
1             H6  56.18333   10.23333  1951  
2            EH4  54.21667 -113.00000  1952  
3    Acapulcoite  16.88333  -99.90000  1976  
4             L6 -33.16667  -64.95000  1902  
..           ...       ...        ...   ...  
994           H5  54.50000   35.20000  1807  
995           H6  13.63333   79.41667  1934  
997           H6  -7.08333  111.53333  1869  
998           L5  -6.66667  106.58333  1922  
999          LL6  47.85000   34.76667  1905  

[930 rows x 14 columns]

In [40]:
len(df[df['year']<2000])

930

### Total 930 meteorit fall before year 2000

## 2. Get all the earth meteorites co-ordinates who fell before the year 1970

In [45]:
# location of all the meterotes co-oridinates fall before year 1970
df[df['year']<1970][['geolocation_coordinates_1','geolocation_coordinates_2']]

geolocation_coordinates_1  geolocation_coordinates_2
0                      6.08333                   50.77500
1                     10.23333                   56.18333
2                   -113.00000                   54.21667
4                    -64.95000                  -33.16667
5                     71.80000                   32.10000
..                         ...                        ...
994                   35.20000                   54.50000
995                   79.41667                   13.63333
997                  111.53333                   -7.08333
998                  106.58333                   -6.66667
999                   34.76667                   47.85000

[781 rows x 2 columns]

## 3. Assuming that the mass of the earth meteorites was in kg, get all those whose mass was more than 10000kg

In [48]:
df[df['mass']>10000]

id  :@computed_region_cbhk_fwbd        name  \
2        6                          NaN        Abee   
7      392                          NaN        Agen   
11     424                          NaN         Aïr   
16     433                          NaN     Akyumak   
27     466                          NaN  Alfianello   
..     ...                          ...         ...   
991  23989                          NaN   Tieschitz   
992  23998                         34.0      Tilden   
994  24004                          NaN    Timochin   
997  24011                          NaN       Tjabe   
998  24012                          NaN    Tjerebon   

     :@computed_region_nnqa_25f4  fall  geolocation_coordinates_1  \
2                            NaN  Fell                 -113.00000   
7                            NaN  Fell                    0.61667   
11                           NaN  Fell                    8.38333   
16                           NaN  Fell                   42.81667   
27                           NaN  Fell                   10.15000   
..                           ...   ...                        ...   
991                          NaN  Fell                   17.11667   
992                       1762.0  Fell                  -89.68333   
994                          NaN  Fell                   35.20000   
997                          NaN  Fell                  111.53333   
998                          NaN  Fell                  106.58333   

     geolocation_coordinates_2 geolocation_type      mass nametype   recclass  \
2                     54.21667            Point  107000.0    Valid        EH4   
7                     44.21667            Point   30000.0    Valid         H5   
11                    19.08333            Point   24000.0    Valid         L6   
16                    39.91667            Point   50000.0    Valid  Iron, IVA   
27                    45.26667            Point  228000.0    Valid         L6   
..                         ...              ...       ...      ...        ...   
991                   49.60000            Point   28000.0    Valid     H/L3.6   
992                   38.20000            Point   74800.0    Valid         L6   
994                   54.50000            Point   65500.0    Valid         H5   
997                   -7.08333            Point   20000.0    Valid         H6   
998                   -6.66667            Point   16500.0    Valid         L5   

       reclat    reclong  year  
2    54.21667 -113.00000  1952  
7    44.21667    0.61667  1814  
11   19.08333    8.38333  1925  
16   39.91667   42.81667  1981  
27   45.26667   10.15000  1883  
..        ...        ...   ...  
991  49.60000   17.11667  1878  
992  38.20000  -89.68333  1927  
994  54.50000   35.20000  1807  
997  -7.08333  111.53333  1869  
998  -6.66667  106.58333  1922  

[243 rows x 14 columns]

In [49]:
len(df[df['mass']>10000])

243

#### Total 243 meteorites fall having mass more than 10000 Kg